src/mds/mdstypes.cc

```diff
@@ -440,29 +440,60 @@ void feature_bitset_t::print(ostream& out) const
   out.flags(f);
 }
 
+/*
+ * metric_spec_t
+ */
+void metric_spec_t::encode(bufferlist& bl) const {
+  using ceph::encode;
+  ENCODE_START(1, 1, bl);
+  encode(metric_flags, bl);
+  ENCODE_FINISH(bl);
+}
+
+void metric_spec_t::decode(bufferlist::const_iterator &p) {
+  using ceph::decode;
+  DECODE_START(1, p);
+  decode(metric_flags, p);
+  DECODE_FINISH(p);
+}
+
+void metric_spec_t::dump(Formatter *f) const {
+  f->dump_stream("flags") << metric_flags;
```

Comment:
`f->dump_object("flags", metric_flags);`

Reply:
I see now that `feature_bitset_t::print` does not yet exist. Please add a method which does this `dump_stream` as here.

Reply:
`feature_bitset_t::print` exists -- `feature_bitset_t::dump` does not. want to add that?

Reply:
btw, `feature_bitset_t::dump` would end up calling `feature_bitset_t::print` since the bitvector needs to transformed to a bitmap value. callers should probably just do a `dump_stream` for `feature_bitset_t` types?

Reply:
added `feature_bitset_t::dump()` w/ `dump_object()` changes -- this dumps as per:

```
        "client_metadata": {
            "client_features": {
                "feature_bits": "0000000000003fff"
            },
            "metric_spec": {
                "metric_flags": {
                    "feature_bits": "0000000000000007"
                }
            },
            "ceph_sha1": "716c628192d7664f9037d1e5eb1acab55de6313c",
            "ceph_version": "ceph version 15.0.0-8435-g716c628192 (716c628192d7664f9037d1e5eb1acab55de6313c) octopus (dev)",
            "entity_id": "x",
            "hostname": "cephalopod-octo",
            "pid": "32265",
            "root": "/"
        }
```

looks ok? (the `metric_flags` seen above is a test val).

Reply:
@batrick ^^

Reply:
LGTM!